components/NewNFTBook.vue

```diff
@@ -966,16 +1027,33 @@ function handleDeliveryMethodChange (value: string) {
   }
 }
 
-function updateClassId ({ classId: newClassId, nftMintCount }) {
-  if (!isStandalonePage.value) {
-    classId.value = newClassId
-    classIdInput.value = newClassId
-    if (nftMintCount) {
-      prices.value[0].stock = nftMintCount
-    }
+async function updateClassId ({ classId: newClassId, nftMintCount }) {
+  classId.value = newClassId
+  classIdInput.value = newClassId
+  if (nftMintCount) {
+    prices.value[0].stock = nftMintCount
+  }
+  if (iscnId.value) {
+    iscnId.value = await getIscnId()
   }
 }
 
+async function getIscnId () {
+  if (sessionData.value?.iscnRecord?.iscnId) {
+    return sessionData.value?.iscnRecord?.iscnId
+  }
```

Comment:
Wouldn't `sessionData` caches the `iscnId` we want to refresh when we call `updateClassId`?

Reply:
[🎨 Invert condition to check for ISCN ID existence before fetching](https://github.com/likecoin/nft-book-press/pull/332/commits/146ae9c98fc8dc49c0cc3ce6a6cdc62211cb0fdc)